doc/source/whatsnew/v1.3.0.rst

```diff
@@ -613,6 +613,7 @@ Other API changes
 - Partially initialized :class:`CategoricalDtype` (i.e. those with ``categories=None`` objects will no longer compare as equal to fully initialized dtype objects.
 - Accessing ``_constructor_expanddim`` on a :class:`DataFrame` and ``_constructor_sliced`` on a :class:`Series` now raise an ``AttributeError``. Previously a ``NotImplementedError`` was raised (:issue:`38782`)
 - Added new ``engine`` and ``**engine_kwargs`` parameters to :meth:`DataFrame.to_sql` to support other future "SQL engines". Currently we still only use ``SQLAlchemy`` under the hood, but more engines are planned to be supported such as ``turbodbc`` (:issue:`36893`)
+- :class:`StringArray` now accepts nan-likes(``None``, ``nan``, ``NaT``, ``NA``, Decimal("NaN")) in its constructor in addition to strings.
```

Comment:
do we need a release note? I'm not sure that users should be constructing a StringArray  directly. For the ArrowStringArray, we are trying to ensure that StringArray and ArrowStringArray behave the same from a user perspective, the array constructor is the one area where these are different.

Reply:
I think this is public API, so it might be necessary.

Reply:
you're right. The recommended function for creating a StringArray is pd.array, but StringArray is public, so I guess we need a release note.